felix/fv/rpf_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build fvtests
+
+package fv_test
+
+import (
+	"fmt"
+	"regexp"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	log "github.com/sirupsen/logrus"
+
+	api "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+
+	. "github.com/projectcalico/calico/felix/fv/connectivity"
+	"github.com/projectcalico/calico/felix/fv/infrastructure"
+	"github.com/projectcalico/calico/felix/fv/utils"
+	"github.com/projectcalico/calico/felix/fv/workload"
+	"github.com/projectcalico/calico/libcalico-go/lib/apiconfig"
+	client "github.com/projectcalico/calico/libcalico-go/lib/clientv3"
+)
+
+var _ = infrastructure.DatastoreDescribe(
+	"RPF tests",
+	[]apiconfig.DatastoreType{apiconfig.Kubernetes},
+	func(getInfra infrastructure.InfraFactory) {
+		var (
+			infra        infrastructure.DatastoreInfra
+			felixes      []*infrastructure.Felix
+			options      infrastructure.TopologyOptions
+			calicoClient client.Interface
+			w            *workload.Workload
+			cc           *Checker
+		)
+
+		BeforeEach(func() {
+			infra = getInfra()
+			options = infrastructure.DefaultTopologyOptions()
+		})
+
+		AfterEach(func() {
+			infra.Stop()
+		})
+
+		JustBeforeEach(func() {
+			felixes, calicoClient = infrastructure.StartNNodeTopology(1, options, infra)
+
+			wIP := "10.65.0.2"
+			w = workload.Run(felixes[0], "w0", "default", wIP, "8055", "udp")
+			w.WorkloadEndpoint.Labels = map[string]string{"name": w.Name}
+			w.ConfigureInInfra(infra)
+
+			pol := api.NewGlobalNetworkPolicy()
+			pol.Namespace = "fv"
+			pol.Name = "policy-1"
+			pol.Spec.Ingress = []api.Rule{{Action: "Allow"}}
+			pol.Spec.Egress = []api.Rule{{Action: "Allow"}}
+			pol.Spec.Selector = "all()"
+
+			pol = func(policy *api.GlobalNetworkPolicy) *api.GlobalNetworkPolicy {
+				log.WithField("policy", dumpResource(policy)).Info("Creating policy")
+				policy, err := calicoClient.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+				Expect(err).NotTo(HaveOccurred())
+				return policy
+			}(pol)
```

Comment:
That's a curious way of writing things - is there a clever reason?

Reply:
nope, me just taking a copy-pasta shortcut, will  redo it.